framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -499,11 +530,12 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 	}
 
 	private async _submitCCUs(ccuParams: Buffer[]): Promise<void> {
-		// This can be changed based on mainchain/sidechain
-		const activeAPIClient = this._sidechainAPIClient;
+		const activeAPIClient = this._sendingChainClient;
 		const activePrivateKey = this._privateKey;
 		const activePublicKey = ed.getPublicKeyFromPrivateKey(activePrivateKey);
-		const activeTargetCommand = COMMAND_NAME_SUBMIT_SIDECHAIN_CCU;
+		const activeTargetCommand = this._isReceivingChainIsMainchain
+			? COMMAND_NAME_SUBMIT_MAINCHAIN_CCU
```

Comment:
Maybe it's just a naming thing, but shouldn't we submit sidechainCCU to the mainchain and vise-versa because we are updating information about sidechain on mainchain

Reply:
yeah it looks confusing but when the receiving chain of the CCU is mainchain then we use `COMMAND_NAME_SUBMIT_MAINCHAIN_CCU `

> MainchainCrossChainUpdate
> This command is used to submit a cross-chain update on the mainchain.

Reply:
ah but this reminds me that we need to also update the API client accordingly
LINE533: `const activeAPIClient = this._sendingChainClient;`